### ISSUE TITLE
About loss function different train_batch outputs

### ISSUE THREAD

**Issue by iceberg2271** [keras-team/keras#14110]
**System information**  
- Have I written custom code (as opposed to using example directory):  
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Linux ubuntu 18.04
- TensorFlow backend (yes / no):  yes
- TensorFlow version:  2.1.0
- Keras version:  2.3.1
- Python version:  3.6.5
- CUDA/cuDNN version:  7.6.5/1
- GPU model and memory:  11GB

You can obtain the TensorFlow version with:  
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"  
You can obtain the Keras version with:  
python -c 'import keras as k; print(k.__version__)'  

**Describe the current behavior**  
this is model compile:
```
                self.dis_out.compile(loss="binary_crossentropy", optimizer=tf.keras.optimizers.Adam(0.0002), metrics=[tf.keras.metrics.binary_accuracy])
                fake = np.zeros((batch_size, 1)).astype('float32')
                noise = np.random.normal(0, 1, (batch_size, 1000))
                fake_img = self.gen.predict(noise)
                metrics_fake = self.dis_out.train_on_batch(fake_img, fake)
                print(metrics_fake)
                print(tf.keras.metrics.binary_accuracy(self.dis_out(fake_img), fake))
```
but It's print result is :
```
tf.Tensor(
[0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.
 0. 0. 0. 0. 0. 0.], shape=(30,), dtype=float32)
[4.4760876e-05, 1.0]
```
It's amazing, and dis_out model output is y_pred:
```
[[0.26062244]
 [0.23994032]
 [0.26172706]
 [0.23823579]
 [0.24264398]
 [0.2332913 ]
 [0.24136345]
 [0.24536803]
 [0.2285582 ]
 [0.22350116]
 [0.24174117]
 [0.25887084]
 [0.23899779]
 [0.25703514]
 [0.22548303]
 [0.22852476]
 [0.23578092]
 [0.22948748]
 [0.2255757 ]
 [0.24915552]
 [0.24830094]
 [0.24881604]
 [0.23851432]
 [0.24653432]
 [0.23793924]
 [0.23045844]
 [0.23743139]
 [0.23210888]
 [0.24127196]
 [0.23754273]]
```

**Describe the expected behavior**  
so,I think the metrics_fake[1] is accuracy,should sample as the metrics.binary_accuracy outputs

**Code to reproduce the issue**  
Provide a reproducible test case that is the bare minimum necessary to generate the problem.  

**Other info / logs**  
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by sushreebarsa** [comment]
@iceberg2271 
In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!
